UI/src/app/components/panel/panel-table/panel-table.component.ts

```diff
@@ -217,23 +217,25 @@ export class PanelTableComponent extends BaseComponent
   }
 
   // connectortype is not reflected in the table
-  reflectConnectorsPin(target: ITable) {
+  reflectConnectorsPin(table: ITable) {
     this.connectortype = {};
     Object.values(this.bridgeService.arrowsCache)
       .filter(connection => {
-        return this.equals(connection.target.tableName, target.name);
+        return this.equals(connection[table.area].tableName, table.name);
       })
       .forEach(connection => {
-        this.showConnectorPinElement(connection, Area.Target);
+        this.showConnectorPinElement(connection, table.area);
       });
   }
 
   showConnectorPinElement(connection: IConnection, area: Area) {
     const rowId = connection[area].name;
     const element = document.getElementById(rowId);
```

Comment:
Is it possible to refactor this and use ViewChild instead of document traverse?

Reply:
it is better to use getElementById here